src/preprocessing/point_in_poly/hierarchical_winding.jl

```diff
@@ -0,0 +1,239 @@
+function determine_closure!(bounding_box, mesh::TriangleMesh{3}, faces, count_directed_edge)
+    (; edge_vertices_ids, face_vertices_ids, face_edges_ids, vertices) = mesh
+    (; min_corner, max_corner) = bounding_box
+
+    count_directed_edge .= 0
+    intersecting_faces = Int[]
+
+    # Find all exterior edges
+    for face in faces
+        v1 = face_vertices_ids[face][1]
+        v2 = face_vertices_ids[face][2]
+        v3 = face_vertices_ids[face][3]
+
+        if in_bounding_box(vertices[v1], min_corner, max_corner) &&
+           in_bounding_box(vertices[v2], min_corner, max_corner) &&
+           in_bounding_box(vertices[v3], min_corner, max_corner)
+            # Face is completely inside the bounding box
+
+            edge1 = face_edges_ids[face][1]
+            edge2 = face_edges_ids[face][2]
+            edge3 = face_edges_ids[face][3]
+
+            if edge_vertices_ids[edge1] == (v1, v2)
+                count_directed_edge[edge1] += 1
+            else
+                count_directed_edge[edge1] -= 1
+            end
+            if edge_vertices_ids[edge2] == (v2, v3)
+                count_directed_edge[edge2] += 1
+            else
+                count_directed_edge[edge2] -= 1
+            end
+            if edge_vertices_ids[edge3] == (v3, v1)
+                count_directed_edge[edge3] += 1
+            else
+                count_directed_edge[edge3] -= 1
+            end
+
+        elseif !in_bounding_box(vertices[v1], min_corner, max_corner) ||
+               !in_bounding_box(vertices[v2], min_corner, max_corner) ||
+               !in_bounding_box(vertices[v3], min_corner, max_corner)
+            # Face is intersecting the boundaries
+
+            push!(intersecting_faces, face)
+        end
+    end
+
+    closing_edges = findall(!iszero, count_directed_edge)
+    resize!(bounding_box.closing_faces, 0)
+
+    if !isempty(closing_edges)
+        closing_vertex = edge_vertices_ids[closing_edges[1]][2]
+    end
+
+    for edge in closing_edges
+        v1 = edge_vertices_ids[edge][1]
+        v2 = edge_vertices_ids[edge][2]
+
+        if v1 == closing_vertex || v2 == closing_vertex
+            continue
+        end
+
+        if count_directed_edge[edge] < 0
+            @inbounds for _ in 1:abs(count_directed_edge[edge])
+                push!(bounding_box.closing_faces, (v1, v2, closing_vertex))
+            end
+        elseif count_directed_edge[edge] > 0
+            @inbounds for _ in 1:abs(count_directed_edge[edge])
+                push!(bounding_box.closing_faces, (v2, v1, closing_vertex))
+            end
+        end
+    end
+
+    for face in intersecting_faces
+        v1 = face_vertices_ids[face][1]
+        v2 = face_vertices_ids[face][2]
+        v3 = face_vertices_ids[face][3]
+
+        # Flip order of vertices
+        push!(bounding_box.closing_faces, (v3, v2, v1))
+    end
+
+    return bounding_box
+end
+
+function in_bounding_box(mesh, faces, min_corner, max_corner)
```

Comment:
```suggestion
function faces_in_box(mesh, faces, min_corner, max_corner)
```

Reply:
Hm I overload the function and use the method also to check if the barycenter is in the box.

Reply:
How about `is_in_box` ?

Reply:
Sounds good to me.